### ISSUE TITLE
Customize component heading level and remove extra empty lines

### ISSUE THREAD

**Issue by mschoettle** [peschmae/glab-component-generator#2]
Thanks a lot for fixing the issues described in #1! I just tested it along with headers and footers and it works great!

Currently, I have a project called `ci-templates` and the README at the project root contains general explanations on how to include templates and components, lists templates and then the components.

For the component, I currently link to the source and provide a description of the component. Following that comes to the input table.

The use of the header and footer works great. There's two small issues I have:

I would like to have the components as a level 3 heading because I have a level 2 heading for templates and one for components, would it be possible to configure this?

The other issue is that all my files have an empty line (newline) at the end which causes redundant newlines after the header, after the component header, and after the component (before the footer). Would it be possible to remove them?

I also have two small ideas, and will just note them here instead of bombarding you with more issues (I can also create them separately if you prefer):

* add another column in the table called "Required" which denotes whether an input is required. For example, I have inputs with `default: ""` which from the output IMHO is not clear that they are optional
* add a snippet with the YAML to include this component somewhere

**Comment by peschmae** [comment]
I've published a new release, adding a `component-header-level` flag to set the header level.

Additionally I've also cleanup some whitespace in the tempalte, especially around the HEADER/FOOTER files.

For the required column I've opened #3

Regarding the YAML snippet, I currently maintain that as part of the HEADER file on each component, and include the required fields there as well. eg.

```yaml
include:
  - component: gitlab.com/<orga>/<project>/kaniko-build-publish@<VERSION>
    inputs:
      image_name: <YOUR_FANCY_IMAGE>
      image_tag: <SOMETHING_USEFUL_NOT_LATEST>
```
